### PR TITLE
Add storybooks, visual testing with chromatic

### PR DESCRIPTION
**Pull request by zephraph**
We've tried to tackle visual testing a few different times, but I thought I'd give it another go. 

I'm integrating a service called [chromatic](https://www.chromaticqa.com) which is a visual testing service build on top of storybooks. That team is the lead on storybooks dev. 

This obviously necessitates adding storybooks to palette, but it's something I've wanted to do for a bit anyway. Even though the gatsby site is a good proxy for a dev environment, it's not quite a development sandbox. Using storybook as a dev/test only space gives us the benefit of display a component in all its variations or other setups that may look messy or be necessary for the docs.

### CODE REVIEW

**Comment by artsyit** [comment]
Deploy preview for *artsy-palette* ready!

Built with commit 1141658acc0a7f37bb46251b58f40116a1b3823e

https://deploy-preview-398--artsy-palette.netlify.com

**Comment by artsyit** [comment]
Deploy preview for *artsy-palette* ready!

Built with commit 6ab4e6f8748a532ae7a6c23cc6284e851507b4b4

https://deploy-preview-398--artsy-palette.netlify.com

File: package.json [hunk 1]
```diff
@@ -7,7 +7,9 @@
     "lint": "yarn workspaces run lint",
     "compile-palette": "yarn workspace @artsy/palette compile",
     "type-check": "yarn compile-palette && yarn workspaces run type-check",
-    "build-docs": "yarn compile-palette && yarn workspace artsy-palette-docs build"
+    "build-docs": "yarn compile-palette && yarn workspace artsy-palette-docs build",
+    "storybook": "yarn workspace @artsy/palette storybook",
```
Comment (damassi) [inline package.json]: Could be a follow-up, can we update the README / docs with notes on this new change / workflow?

File: packages/palette/package.json [hunk 2]
```diff
@@ -26,7 +26,10 @@
     "type-check": "tsc --emitDeclarationOnly --pretty",
     "type-declarations": "tsc --emitDeclarationOnly",
     "watch": "concurrently --raw --kill-others 'yarn compile -w' 'yarn type-declarations -w'",
-    "watch:emission": "yarn clean:emission && concurrently --raw --kill-others 'yarn compile:emission -w' 'yarn type-declarations -w --outDir ../../../emission/node_modules/@artsy/palette/dist'"
+    "watch:emission": "yarn clean:emission && concurrently --raw --kill-others 'yarn compile:emission -w' 'yarn type-declarations -w --outDir ../../../emission/node_modules/@artsy/palette/dist'",
+    "storybook": "start-storybook -p 6006",
+    "build-storybook": "build-storybook",
```
Comment (damassi) [inline packages/palette/package.json]: Is this accessible via netlify too?

Comment (zephraph) [reply packages/palette/package.json]: It's not enabled right now. Chromatic gives a storybook preview of every deploy though, so we could probably lean on that.

**Comment by orta** [comment]
Yah, looks good to me - think `yarn start` should probably do both too

**Comment by zephraph** [comment]
Let me do the doc updates as a part of the PR and the general communication before we merge. 

I don't want to thrash people's workflow without clear communication.

**Comment by artsyit** [comment]
:rocket: PR was released in v4.11.0 :rocket:

**Pull request merged**